frontend/src/components/molecules/AddTransaction.tsx

```diff
@@ -1,15 +1,67 @@
 import moment from 'moment';
-import React from 'react';
+import React, { useReducer, ReactElement } from 'react';
 import styled from 'styled-components';
 import { useAuthState } from '../../store/contexts/auth';
 import { useTransactionDispatch } from '../../store/contexts/transactions';
 import { TransactionActions } from '../../store/reducers/transactions';
 import Collapsable from '../atoms/Collapsable';
 import Form from './Form';
+import SnackBarContainer from '../atoms/SnackBarContainer';
+
+const initialState = { snax: <div></div>, content: '' };
```

Comment:
Storing a react element in a store like this is generally not recommended, as it has to recreate this component every time the reducer updates. I must admit I've never seen this done before, so +1 for creativity, but this should probably be changed. I believe there are quite some downsides to doing it like this, especially considering how mounting this onto the shadow-dom and dom would be done.